utils/build-presets.ini

```diff
@@ -2365,6 +2365,30 @@ mixin-preset=stdlib_DA_standalone,build
 test
 validation-test
 
+[preset: stdlib_S_standalone_minimal,build]
+mixin-preset=stdlib_base_standalone
+
+build-subdir=stdlib_S_standalone
```

Comment:
Some thoughts:

1. We probably want to have a couple of different presets here. Specifically, on CI we definitely want to compile with asserts enabled. At least in the compiler.
2. Can you split out the non-specific stuff into a base preset that is stdlib_S_standalone,build and have this inherit from that with the minimal stuff?

Reply:
Makes sense. Will do.

Reply:
Ok, so I added separate 'S' and 'SA' presets that aren't specific, and then a mixin `mixin_stdlib_minimal` which drops certain features we don't want in the "minimal" variant. I plan to add more things into this mixin. Then I've renamed the actual preset to `stdlib_S_standalone_minimal_macho_x86_64`, because that's what it is: It's making a concrete build of this variant for x86_64+Mach-O. This way, someone can add an ELF concrete build if they want.